src/sys/stm_tests.ml

```diff
@@ -0,0 +1,242 @@
+open QCheck
+open STM
+
+module SConf =
+struct
+  type cmd =
+    | File_exists of string list
+    | Mkdir of string list * string * int
+    | Rmdir of string list * string
+    | Readdir of string list
+    | Touch of string list * string * int
+    [@@deriving show { with_path = false }]
+
+  module Map_names = Map.Make (String)
+
+  type filesys =
+    | Directory of {perm: int; fs_map: filesys Map_names.t}
+    | File of {perm: int}
+
+  type state = filesys
+
+  type sut   = unit
+
+  let (/) = Filename.concat
+
+  let update_map_name map_name k v = Map_names.add k v (Map_names.remove k map_name)
+
+  let arb_cmd _s  =
+    let name_gen = Gen.(oneofl ["aaa" ; "bbb" ; "ccc" ; "ddd" ; "eee"]) in
+    let path_gen = Gen.map (fun path -> path) (Gen.list_size (Gen.int_bound 5) name_gen) in (* can be empty *)
+    let perm_gen = Gen.(oneofl [0o777]) in
```

Comment:
`Gen.return 0o777` should give you a faster, constant generator

Reply:
Good point !